src/ports/catalog/queries.ts

```diff
@@ -0,0 +1,355 @@
+import SQL, { SQLStatement } from 'sql-template-strings'
+import {
+  CatalogFilters,
+  CatalogQueryFilters,
+  CatalogSortBy,
+  CatalogSortDirection,
+} from './types'
+import {
+  EmoteCategory,
+  EmotePlayMode,
+  GenderFilterOption,
+  NFTCategory,
+  WearableCategory,
+  WearableGender,
+} from '@dcl/schemas'
+import { FragmentItemType } from '../items/types'
+
+export const getLatestSubgraphSchema = (subgraphName: string) =>
+  SQL`
+    SELECT 
+        entity_schema 
+    FROM 
+        satsuma.subgraph_schema 
+    WHERE 
+        satsuma_subgraph_name = ${subgraphName}
+    `
+
+export function getOrderBy(filters: CatalogFilters) {
+  const { sortBy, sortDirection, onlyListing, isOnSale } = filters
+  const sortByParam = sortBy ?? CatalogSortBy.NEWEST
+  const sortDirectionParam = sortDirection ?? CatalogSortDirection.ASC
+
+  // When seeing "Not for sale", the only sort available is the Newest one
+  if (isOnSale === false) {
+    if (sortByParam === CatalogSortBy.NEWEST) {
+      return `ORDER BY items.first_listed_at ${sortDirectionParam}\n`
```

Comment:
I think this is already contemplated a few lines below in the `case CatalogSortBy.NEWEST:` (line 46).

We may just return an empty string if the `isOnSale` filter is `false` and the `sortByParam` is not `NEWEST`. WDYT?

```ts
if (isOnSale === false && sortByParam !== CatalogSortBy.NEWEST) return ''
```